integration/tests/cook/test_basic.py

```diff
@@ -2639,6 +2698,32 @@ def test_kubernetes_disallowed_volumes(self):
         finally:
             util.kill_jobs(self.cook_url, [job_uuid])
 
+    @unittest.skipUnless(util.using_kubernetes(), 'Test requires kubernetes')
+    def test_kubernetes_disallowed_var_names(self):
+        settings = util.settings(self.cook_url)
+        bad_var_name = 'BADVAR'
+        disallowed_container_paths = util.get_in(settings, 'kubernetes', 'disallowed-var-names')
+        if disallowed_container_paths is None or bad_var_name not in disallowed_container_paths:
+            self.skipTest(f'Requires {bad_var_name} to be in the disallowed-var-names config')
+        docker_image = util.docker_image()
+
+        container = {'type': 'docker',
+                     'docker': {'image': docker_image,
+                                'network': 'HOST',
+                                'force-pull-image': False,
+                                'parameters': [{'key': 'env',
+                                                'value': f'{bad_var_name}=set'}]}}
+        command = 'bash -c \'if [ -z ${BADVAR+x} ]; then exit 0; else exit 1; fi\''
```

Comment:
What does `BADVAR+x` do?

Reply:
That is equivalent to x if `BADVAR` is set to anything (including empty string) and still null if it is not set. It seems like the best way to test for a variable not being set.